src/Symfony/Component/Messenger/Bridge/Doctrine/Transport/Connection.php

```diff
@@ -510,4 +541,53 @@ private function fallBackToForUpdate(QueryBuilder $query, string $sql): string
             return $sql;
         }
     }
+
+    /**
+     * @return int[]
+     */
+    public function extractDoctrineEnvelopeListIds(array $doctrineEnvelopeList): array
+    {
+        return array_map(
```

Comment:
`return array_column($doctrineEnvelopeList, 'id')` will do the same and might be a bit faster by keeping all the logic in the C layer.